include/quic/connection_ids.hpp

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <stdexcept>
+
+extern "C"
+{
+#include <gnutls/crypto.h>
+}
+
+#include "types.hpp"
+#include "utils.hpp"
+
+namespace oxen::quic
+{
+    // Unique reference ID for each individual connection. QUIC allows for a
+    // connection to be reached at any of n number of connection ID's (see above
+    // constexpr) at a time. As a result, we key each of those QUIC CID's to the
+    // unique reference ID, to which the connection shared_ptr is keyed in the
+    // master map container
+    struct alignas(uint64_t) ReferenceID final
+    {
```

Comment:
The name here felt a little bit odd to use in storage server, because I have a "oxen::quic::ReferenceID".  Perhaps renaming to "ConnRefID" or something along those lines to indicate that it is specific to connections would be useful?

Reply:
I suppose it could be `oxen::quic::connection_interface::ReferenceID` but that brings up memory of Java.

Reply:
From discussion:
- rename `ConnectionID` to `quic_cid` (this is an internal libquic type)
- rename `ReferenceID` to `ConnectionID` for (mostly) external use